WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -541,9 +541,9 @@ public IEnumerable<HdPubKey> GetKeys(KeyState? keyState = null, bool? isInternal
 			}
 		}
 
-		public int CountConsecutiveCleanKeys(bool isInternal)
+		public int CountConsecutiveUnusedKeys(bool isInternal)
 		{
-			var keyIndexes = GetKeys(KeyState.Clean, isInternal).Select(x => x.Index).ToArray();
+			var keyIndexes = GetKeys(x => x.IsInternal == isInternal && x.KeyState != KeyState.Used).OrderBy(x => x.Index).Select(x => x.Index).ToArray();
```

Comment:
Could you check if this is better in any way?
```suggestion
			var keyIndexes = GetKeys(x => x.IsInternal == isInternal && x.KeyState != KeyState.Used).Select(x => x.Index).OrderBy(x => x).ToArray();
```

Reply:
@nopara73

Reply:
More readable.